### PR TITLE
update sphinx and theme

### LINKED ISSUE #275: update sphinx and theme
Other documentation projects have switched to the [pydata_sphinx_theme](https://pydata-sphinx-theme.readthedocs.io/).  Let's do that here and also update the sphinx configuration in [`docs/source/conf.py`](https://github.com/bluesky/hklpy/blob/main/docs/source/conf.py).

After #267.

### LINKED ISSUE #278: When to publish the docs?
@prjemian, do you intend to only publish the docs manually via `workflow_dispatch`?
I don't see where this value would be set to `true` to include the steps on release/etc.

_Originally posted by @mrakitin in https://github.com/bluesky/hklpy/pull/276#discussion_r1367757050_

### PR DESCRIPTION
**Pull request by prjemian**
- close #275
- close #278

### CODE REVIEW

**Comment by prjemian** [comment]
This branch needs #282 to view the docs built in each workflow.

**Comment by prjemian** [comment]
Odd failure in Py3.10 testing CI:

```py
Downloading apischema-0.18.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
   ━━━━━━━━━━━━━━━━━━━━                     2.1/4.1 MB 509.9 kB/s eta 0:00:04
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    unknown package:
        Expected sha256 a62348374a42c47e311fa3a8945b3c54bf4b52029eda2c819be0a1b63280b04d
             Got        50738c1a80b0174a5b[33](https://github.com/bluesky/hklpy/actions/runs/6739703235/job/18321649609?pr=285#step:5:34)22ecf32b5ed6085b52198def54[34](https://github.com/bluesky/hklpy/actions/runs/6739703235/job/18321649609?pr=285#step:5:35)d0475eefbff7ab54
```

File: docs/source/index.rst [hunk 1]
```diff
@@ -51,3 +49,5 @@ Indices
 
 * :ref:`General Index <genindex>`
 * :ref:`Index of Modules <modindex>`
+
+.. note:: *hklpy* documentation built |today| for version |version|.
```
Comment (mrakitin) [inline docs/source/index.rst:53]: 👍

**Review by mrakitin — APPROVED** [review]
Looks good! This is a more modern theme.

**Pull request merged**